Fix `mypy` warning for `narwhals>=1.27.0`

Very minor change required after `narwhals`'s next release.
We've got an early warning thanks to their CI 🙂 

```py
altair/utils/data.py:366: error: Argument 1 to "sanitize_geo_interface" has
incompatible type "MutableMapping[Any, Any] | Any | Series[Any]"; expected
"MutableMapping[Any, Any]"  [arg-type]
        return sanitize_geo_interface(data.__geo_interface__)
                                      ^~~~~~~~~~~~~~~~~~~~~~
Found 1 error in 1 file (checked 383 source files)
```

# Related
- https://github.com/narwhals-dev/narwhals/releases/tag/v1.27.0
- https://github.com/narwhals-dev/narwhals/pull/2027
  - https://github.com/narwhals-dev/narwhals/pull/2027#issuecomment-2661641699
  - https://github.com/narwhals-dev/narwhals/pull/2027#pullrequestreview-2620583391
  - https://github.com/narwhals-dev/narwhals/actions/runs/13358588562/job/37304764493?pr=2027
- https://github.com/vega/altair/pull/3550#discussion_r1725283165

--- a/altair/utils/core.py
+++ b/altair/utils/core.py
@@ -38,6 +38,8 @@ if TYPE_CHECKING:
     from altair.utils._dfi_types import DataFrame as DfiDataFrame
     from altair.vegalite.v5.schema._typing import StandardType_T as InferredVegaLiteType
 
+    _PandasDataFrameT = TypeVar("_PandasDataFrameT", bound="pd.DataFrame")
+
 TIntoDataFrame = TypeVar("TIntoDataFrame", bound=IntoDataFrame)
 T = TypeVar("T")
 P = ParamSpec("P")
@@ -328,7 +330,7 @@ def numpy_is_subtype(dtype: Any, subtype: Any) -> bool:
         return False
 
 
-def sanitize_pandas_dataframe(df: pd.DataFrame) -> pd.DataFrame:  # noqa: C901
+def sanitize_pandas_dataframe(df: _PandasDataFrameT) -> _PandasDataFrameT:  # noqa: C901
     """
     Sanitize a DataFrame to prepare it for serialization.
 
@@ -351,7 +353,7 @@ def sanitize_pandas_dataframe(df: pd.DataFrame) -> pd.DataFrame:  # noqa: C901
     import numpy as np
     import pandas as pd
 
-    df = df.copy()
+    df = cast("_PandasDataFrameT", df.copy())
 
     if isinstance(df.columns, pd.RangeIndex):
         df.columns = df.columns.astype(str)

--- a/altair/utils/data.py
+++ b/altair/utils/data.py
@@ -350,7 +350,6 @@ def _compute_data_hash(data_str: str) -> str:
     return hashlib.sha256(data_str.encode()).hexdigest()[:32]
 
 
-# ahhhhhh
 def _from_geo_interface(data: SupportsGeoInterface) -> dict[str, Any]:
     """
     Santize a ``__geo_interface__`` w/ pre-santize step for ``pandas`` if needed.

--- a/altair/utils/data.py
+++ b/altair/utils/data.py
@@ -350,20 +350,17 @@ def _compute_data_hash(data_str: str) -> str:
     return hashlib.sha256(data_str.encode()).hexdigest()[:32]
 
 
-def _from_geo_interface(data: SupportsGeoInterface | Any) -> dict[str, Any]:
+# ahhhhhh
+def _from_geo_interface(data: SupportsGeoInterface) -> dict[str, Any]:
     """
     Santize a ``__geo_interface__`` w/ pre-santize step for ``pandas`` if needed.
 
-    Notes
-    -----
-    Split out to resolve typing issues related to:
-    - Intersection types
-    - ``typing.TypeGuard``
-    - ``pd.DataFrame.__getattr__``
+    Introduces an intersection type::
+
+        geo: <subclass of SupportsGeoInterface and DataFrame> | SupportsGeoInterface
     """
-    if is_pandas_dataframe(data):
-        data = sanitize_pandas_dataframe(data)
-    return sanitize_geo_interface(data.__geo_interface__)
+    geo = sanitize_pandas_dataframe(data) if is_pandas_dataframe(data) else data
+    return sanitize_geo_interface(geo.__geo_interface__)
 
 
 def _data_to_json_string(data: DataType) -> str:

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -20,7 +20,7 @@ dependencies = [
     # If you update the minimum required jsonschema version, also update it in build.yml
     "jsonschema>=3.0",
     "packaging",
-    "narwhals>=1.26.0"
+    "narwhals>=1.27.1"
 ]
 description = "Vega-Altair: A declarative statistical visualization library for Python."
 readme = "README.md"

